src/components/v2/V2Project/LaunchProjectPayerModal.tsx

```diff
@@ -0,0 +1,147 @@
+import { t, Trans } from '@lingui/macro'
+import { useState } from 'react'
+
+import { TransactionReceipt } from '@ethersproject/providers'
+import { TransactorInstance } from 'hooks/Transactor'
+
+import { Modal } from 'antd'
+import { JBDiscordLink } from 'components/Landing/QAs'
+import EtherscanLink from 'components/shared/EtherscanLink'
+import CopyTextButton from 'components/shared/CopyTextButton'
+
+import { readProvider } from 'constants/readProvider'
+import TransactionModal from '../../shared/TransactionModal'
+
+const DEPLOY_EVENT_IDX = 0
+
+/**
+ * Return the address of the project payer created from a `deployProjectPayer` transaction.
+ * @param txReceipt receipt of `deployProjectPayer` transaction
+ */
+const getProjectPayerAddressFromReceipt = (
+  txReceipt: TransactionReceipt,
+): string => {
+  const newProjectPayerAddress = txReceipt?.logs[DEPLOY_EVENT_IDX]?.address
+  return newProjectPayerAddress
+}
+export default function LaunchProjectPayerModal({
+  visible,
+  onClose,
+  useDeployProjectPayerTx,
+  onConfirmed,
+}: {
+  visible: boolean
+  onClose: VoidFunction
+  useDeployProjectPayerTx: () => TransactorInstance<{}> | undefined
+  onConfirmed?: VoidFunction
+}) {
+  const [loadingProjectPayer, setLoadingProjectPayer] = useState<boolean>()
+  const [transactionPending, setTransactionPending] = useState<boolean>()
+  const [projectPayerAddress, setProjectPayerAddress] = useState<string>()
+
+  const [confirmedModalVisible, setConfirmedModalVisible] = useState<boolean>()
+  // TODO: load project payer and show different thing in this section if the project already has one
+  // (Issue: #897)
+
+  const deployProjectPayerTx = useDeployProjectPayerTx()
+
+  function deployProjectPayer() {
+    if (!deployProjectPayerTx) return
+
+    setLoadingProjectPayer(true)
+
+    deployProjectPayerTx(
+      {
+        args: [],
+      },
+      {
+        onDone() {
+          setTransactionPending(true)
+        },
+        async onConfirmed(result) {
+          const txHash = result?.transaction?.hash
+          if (!txHash) {
+            return
+          }
+
+          const txReceipt = await readProvider.getTransactionReceipt(txHash)
+          const newProjectPayerAddress =
+            getProjectPayerAddressFromReceipt(txReceipt)
+          if (newProjectPayerAddress === undefined) {
+            return
```

Comment:
is this an error case? should we display an error notification to the user if so?

Reply:
Yeah good call. Done